st2common/tests/unit/test_service_setup.py

```diff
@@ -216,3 +217,23 @@ def mock_get_logging_config_path():
             run_migrations=False,
             register_runners=False,
         )
+
+    def test_deregister_service_when_service_registry_enabled(self):
+        service = "api"
+        service_setup.register_service_in_service_registry(
+            service, capabilities={"hostname": "", "pid": ""}
+        )
+        coordinator = coordination.get_coordinator(start_heart=True)
+        members = coordinator.get_members(service.encode("utf-8"))
+        self.assertEqual(len(list(members.get())), 1)
+        service_setup.deregister_service(service)
+        self.assertEqual(len(list(members.get())), 0)
+
+    def test_deregister_service_when_service_registry_disables(self):
+        service = "api"
+        try:
+            service_setup.deregister_service(service)
+        except:
+            assert False, "service_setup.deregister_service raised exception"
```

Comment:
@khushboobhatia01 What happens if this error is raised on shutdown? I think it is ok to make this silent if there is nothing to be deregister.

Reply:
@m4dcoder We don't expect deregister() to raise an exception when service_registry is disabled or there's nothing to deregister. This test case would fail only when some unintended change is causing deregister to fail. All edge cases are handled in deregister(). Therefore it might be good to know if some new change is causing it to fail.